recipes/powershell4.rb

```diff
@@ -30,14 +30,25 @@
     # Ensure .NET 4.5 is installed or installation will fail silently per Microsoft. Only necessary for Windows 2008R2 or 7.
     include_recipe 'ms_dotnet45' if windows_version.windows_server_2008_r2? || windows_version.windows_7?
 
-    # Reboot if user specifies immediate_reboot
-    include_recipe 'powershell::windows_reboot' if node['powershell']['installation_reboot_mode'] == 'immediate_reboot'
+    # Get powershell version
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies doesn't specify no_reboot and powershell version < 4
+    include_recipe 'powershell::windows_reboot' unless node['powershell']['installation_reboot_mode'] == 'no_reboot' && powershell_version < 4
```

Comment:
There is a possibility that the above code may not be required at all now since you are notifying `windows_reboot[powershell]` from `windows_package` recipe now. If that's the case then we should remove the version check code written above after complete testing.